src/core/thread/mle.cpp

```diff
@@ -2260,13 +2260,27 @@ otError Mle::HandleAdvertisement(const Message &aMessage, const Ip6::MessageInfo
 
     otLogInfoMle(GetInstance(), "Received advertisement from %04x", sourceAddress.GetRloc16());
 
+    aMessageInfo.GetPeerAddr().ToExtAddress(macAddr);
+
     if (mRole != OT_DEVICE_ROLE_DETACHED)
     {
-        SuccessOrExit(error = netif.GetMle().HandleAdvertisement(aMessage, aMessageInfo));
+        if (mDeviceMode & ModeTlv::kModeFFD)
```

Comment:
maybe here better use `#if OPENTHREAD_FTD`, or else some FFD checks in `HandleAdvertisement() `of mle_router.cpp should also be cleared up.

Reply:
As discussed offline, let's do clean up in an separate PR.